src/main/java/org/wildfly/security/password/PasswordUtil.java

```diff
@@ -310,6 +376,12 @@ private static StringBuilder getCryptStringToBuilder(Password password) throws I
         return b;
     }
 
+    private static void appendAlgorithmToElytronCryptString(StringBuilder b, String algorithm) {
+        b.append("$ely$");
+        ByteIterator.ofBytes(algorithm.getBytes()).base64Encode().drainTo(b);
```

Comment:
Not really needed. The proposed format uses base64 in all parts, so I use to encode the ```algorithm``` as well.